services/secrets/masterkey.go

```diff
@@ -0,0 +1,27 @@
+// Copyright 2021 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package secrets
+
+import (
+	"fmt"
+)
+
+// ErrMasterKeySealed is returned when trying to use master key that is sealed
+var ErrMasterKeySealed = fmt.Errorf("master key sealed")
+
+// MasterKeyProvider provides master key used for encryption
+type MasterKeyProvider interface {
+	Init() error
+
+	GenerateMasterKey() ([][]byte, error)
```

Comment:
Will there ever be providers which return multiple secrets? If yes, why does the `Unseal` method only take one secret?

Reply:
Yes there could be for example using Shamir's Secret Sharing (https://github.com/lafriks/go-shamir) where master key would be split to multiple parts. I initially wanted to add this also but that would require too many changes so I left it out. It would require gitea to start in unsealed state (similar to install lock = false state) and would switch to normal state only after X parts of Y part secrets are provided to it

Reply:
Only secrets we can not protect even in future by master key are `INTERNAL_TOKEN` and database `PASSWD`

Reply:
> Yes there could be for example using Shamir's Secret Sharing (https://github.com/lafriks/go-shamir) where master key would be split to multiple parts.

Perhaps you should document that inside the function signature to not confuse future developers.